### PR TITLE
GH835: adds CRD update step to helm update instructions

### PR DESCRIPTION
**Pull request by misterjacko**
Product Version(s):

Calico: all versions

Issue:

#835 

Link to docs preview:

https://deploy-preview-836--tigera.netlify.app/calico/latest/operations/upgrading/kubernetes-upgrade#upgrade-from-calico-versions-prior-to-v3230

https://deploy-preview-836--tigera.netlify.app/calico/3.25/operations/upgrading/kubernetes-upgrade#upgrade-from-calico-versions-prior-to-v3230

https://deploy-preview-836--tigera.netlify.app/calico/3.24/operations/upgrading/kubernetes-upgrade#upgrade-from-calico-versions-prior-to-v3230

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [x] Deploy preview inspected wherever changes were made 
- [x] Build completed successfully
- [x] Test have passed

### CODE REVIEW

**Comment by misterjacko** [comment]
@tigera/docs ready for review :-)

**Comment by bmckercher123** [comment]
Thank you for taking the time to submit these updates. Much appreciated. Reviewers have been assigned.

**Comment by bmckercher123** [comment]
@dimitri-nicolo Let us know results of your testing kubectl apply versus replace.

File: calico/operations/upgrading/kubernetes-upgrade.mdx [hunk 1]
```diff
@@ -53,6 +53,12 @@ ownership of the helm resources to the new chart location.
    kubectl patch clusterrolebinding tigera-operator tigera-operator --type=merge -p '{"metadata": {"annotations": {"meta.helm.sh/release-namespace": "tigera-operator"}}}'
    ```
 
+1. Apply the {{version}} CRDs:
+
+   ```bash
+   kubectl replace -f {{manifestsUrl}}/manifests/operator-crds.yaml
```
Comment (dimitri-nicolo) [inline calico/operations/upgrading/kubernetes-upgrade.mdx]: `kubectl replace` will fail to create new resources. If any new CRDs exist in the YAML, then we will see an error similar to:

`Error from server (NotFound): error when replacing "{{manifestsUrl}}/manifests/operator-crds.yaml": customresourcedefinitions.apiextensions.k8s.io "*.crd.projectcalico.org" not found`

One way to ensure that existing CRDs are updated and new ones are created, would be to use:
`kubectl apply --server-side --force-conflicts -f {{manifestsUrl}}/manifests/operator-crds.yaml`

Comment (misterjacko) [reply calico/operations/upgrading/kubernetes-upgrade.mdx]: Thank you @dimitri-nicolo, I will edit and push those changes.  Would that same change be appropriate for the CRD update step in the "All other Upgrades" section?  Happy to change that as well.

Comment (dimitri-nicolo) [reply calico/operations/upgrading/kubernetes-upgrade.mdx]: @misterjacko yes, I believe so. If you could also add those changes, that would be great. Thanks!

**Review by dimitri-nicolo — APPROVED** [review]
lgtm

**Review by bmckercher123 — APPROVED** [review]
LGTM

**Comment by bmckercher123** [comment]
Thank you @misterjacko for a timely turnaround on this PR!

**Pull request merged**